File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/actions/fight/GameActionFightModifyEffectsDurationMessage.scala





















// Generated on 05/03/2013 10:34:34
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GameActionFightModifyEffectsDurationMessage extends AbstractGameActionMessage {
  override val messageId = 6304

  protected trait Message extends AbstractGameActionMessage.Message with Serializable {
    override val messageId = GameActionFightModifyEffectsDurationMessage.messageId

    val targetId: Int
    val delta: Short

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeInt(targetId)
      buf.writeShort(delta)
    }

  }

  protected def create(actionId: Short, sourceId: Int, targetId: Int, delta: Short): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val actionId = buf.readShort()
    val sourceId = buf.readInt()
    val targetId = buf.readInt()
    val delta = buf.readShort()

    create(actionId, sourceId, targetId, delta)
  }

}

object GameActionFightModifyEffectsDurationMessage extends GameActionFightModifyEffectsDurationMessage {
  case class MessageImpl(actionId: Short, sourceId: Int, targetId: Int, delta: Short) extends Message
  type Target = MessageImpl

  override def create(actionId: Short, sourceId: Int, targetId: Int, delta: Short) = MessageImpl(actionId, sourceId, targetId, delta)

  def apply(actionId: Short, sourceId: Int, targetId: Int, delta: Short) = create(actionId, sourceId, targetId, delta)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GameActionFightModifyEffectsDurationMessage.java


// Generated on 05/08/2013 19:37:39
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GameActionFightModifyEffectsDurationMessage extends AbstractGameActionMessage {
    public static final int MESSAGE_ID = 6304;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int targetId;
    public short delta;
    
    public GameActionFightModifyEffectsDurationMessage() { }
    
    public GameActionFightModifyEffectsDurationMessage(short actionId, int sourceId, int targetId, short delta) {
        super(actionId, sourceId);
        this.targetId = targetId;
        this.delta = delta;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeInt(targetId);
        buf.writeShort(delta);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        targetId = buf.readInt();
        delta = buf.readShort();
    }
    
}
